src/core/tsi/transport_security.h

```diff
@@ -88,7 +88,7 @@ struct tsi_handshaker {
 typedef struct {
   tsi_result (*extract_peer)(const tsi_handshaker_result *self, tsi_peer *peer);
   tsi_result (*create_zero_copy_grpc_protector)(
-      const tsi_handshaker_result *self,
+      void *exec_ctx, const tsi_handshaker_result *self,
```

Comment:
It seems really sub-optimal to make this `void`, since the zero-copy frame protector depends on grpc anyway.  I'm guessing that the reason for this is that this TSI handshaker API is used for both grpc and non-grpc uses cases?  Is there any way to work around this, such as maybe using a `typedef` for non-grpc uses that defines `grpc_exec_ctx` as `void`?

Reply:
Yes, the goal is to keep TSI interface (transport_security_interface.h, transport_security.x) independent of grpc. There are other callers of TSI besides grpc.

How about we typedef grpc_exec_ctx, similar to what we did on tsi_zero_copy_grpc_protector on transport_security_interface.h. Just declare it without define it.
https://github.com/grpc/grpc/blob/master/src/core/tsi/transport_security_interface.h#L72

Reply:
Sounds fine to me.  Go ahead and give it a try, and we'll see if anything breaks.

Reply:
It does not work. It breaks asan. I have chatted with Julien. He thinks we need to have tsi_handshaker and tsi_handshaker_result interface independent of grpc. I rolled back to use void in the vtable layer. For zero-copy protector, we will cast to grpc_exec_ctx. What do you think?

Reply:
Note that in tsi_handshaker_result_create_zero_copy_grpc_protector() still pass grpc_exec_ctx. It is just vtable needs to use void to make TSI interface independent.

Reply:
It's not clear to me how this could have caused asan failures.  Do we know how that happened?  I would have thought that it could have caused build failures, but I don't see how it could cause the build to succeed and then leak memory.

If this won't work for some reason, I can live with void, but let's at least add a comment explaining why it's void:

```
// The exec_ctx parameter is supposed to be of type grpc_exec_ctx*, but
// we're using void* instead to avoid making the TSI API depend on grpc.
// The create_zero_copy_grpc_protector() method is only used in grpc, where we
// do need the exec_ctx passed through, but the API still needs to compile
// in other applications, where grpc_exec_ctx is not defined.
```

Reply:
When I tried yesterday, it pass the tests when I ran 
./tools/run_tests/run_tests.py -l c
But when I ran asan test, it complains
./tools/run_tests/run_tests.py -l c -c asan

./src/core/tsi/transport_security.h:88:30: error: redefinition of typedef 'grpc_exec_ctx' is a C11 feature [-Werror,-Wtypedef-redefinition]
typedef struct grpc_exec_ctx grpc_exec_ctx;
                             ^
include/grpc/impl/codegen/exec_ctx_fwd.h:24:30: note: previous definition is here
typedef struct grpc_exec_ctx grpc_exec_ctx;
                             ^
1 error generated.